### PR TITLE
Add a testsuite for CMake

### PR DESCRIPTION
**Pull request by maxGimeno**
## Summary of Changes

- usage from the source code, without configuring CGAL at all, with CGAL_DIR pointing to the directory containing:
 - [x] Git/CGALConfig.cmake (for the Git layout only),
   - [x] CGAL-5.x*/CGALConfig.cmake (from a tarball),
   - [x] Git/Installation/lib/cmake/CGAL/CGALConfig.cmake (Git) or CGAL-5.x*/lib/cmake/CGAL/CGALConfig.cmake (tarball)
- [x] configure all CGAL with -DWITH_examples=ON -DWITH_demos=ON -DWITH_tests=ON, and then launch CTest on a few labels. (Travis only)
 - [x] configure CGAL (as header-only), then use its build-directory as CGAL_DIR (we might want to support that use-case to be compatible with our usage with non-header-only, for CGAL-4.14),
 - CGAL installed in the default prefix ( Travis only):
   - [x] with the CGAL_Qt5 header-files.
   - [x] without the CGAL_Qt5 header-files,
 - CGAL installed in a non-default prefix:
    - [x] with the CGAL_Qt5 header-files.
    - [x] without the CGAL_Qt5 header-files.

### CODE REVIEW

File: Installation/test/Installation/CMakeLists.txt [hunk 1]
```diff
@@ -135,8 +136,144 @@ CGAL_installation_test_find_package_version(equal)
...
+  file(MAKE_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/non_standard_install_qt5)
+  file(MAKE_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/non_standard_build_qt5)
+  file(MAKE_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/test_config_file_qt5)
+  configure_file(test_configuration_qt5.cmake.in ${CMAKE_CURRENT_BINARY_DIR}/test_config_file_qt5/CMakeLists.txt @ONLY)
+endif()#CGAL_Qt5_FOUND
+
+
+#If ctest is ran from a global config, CGAL_SOURCE_DIR exists, but from Installation/test it doesn't. In that case, however, there is a CGAL_DIR.
+if("${CGAL_SOURCE_DIR}" STREQUAL "")
+  set(CGAL_SOURCE_DIR ${CGAL_DIR})
```
[the remark this answers is not in the document]
Comment (lrineau) [reply Installation/test/Installation/CMakeLists.txt:162]: Well, actually, there a lot of similar issues. Absolute paths in CMake should always be quoted.
See https://stackoverflow.com/a/35853080/1728537

[the remark this answers is not in the document]
Comment (maxGimeno) [reply Installation/test/Installation/CMakeLists.txt:162]: You cannot define a XXX_DIR for the config mode with quotes, or they are considered an invalid path and ignored.

**Comment by sloriot** [comment]
I guess responsible for the warnings in Installation package in CGAL-5.1-Ic-214. Labelled as "test failing" but it might be expected as some test platforms might need to be updated.

**Comment by maxGimeno** [comment]
Successfully tested in https://cgal.geometryfactory.com/CGAL/testsuite/results-5.1-Ic-222.shtml

**Comment by lrineau** [comment]
That does not work. In [these results](https://cgal.geometryfactory.com/CGAL/testsuite/CGAL-5.1-Ic-224/Installation/TestReport_lrineau_Debian-Testing.gz), a few tests fail, but the cell is green [in the summary page](https://cgal.geometryfactory.com/CGAL/testsuite/results-5.1-Ic-224.shtml#Installation).

File: Installation/test/Installation/CMakeLists.txt [hunk 2]
```diff
@@ -126,7 +126,7 @@ function(CGAL_installation_test_find_package_version mode)
 
   configure_file(test_find_package.cmake.in ${CMAKE_CURRENT_BINARY_DIR}/test_find_package_version_${mode}/CMakeLists.txt)
   add_test(NAME test_find_package_version_${mode}
-    COMMAND ${CMAKE_COMMAND} -DCMAKE_BUILD_TYPE=Release ${CMAKE_CURRENT_BINARY_DIR}/test_find_package_version_${mode}
+    COMMAND ${CMAKE_COMMAND} $ENV{CMAKE_GENERATOR} -C$ENV{INIT_FILE} -DCMAKE_BUILD_TYPE=Release ${CMAKE_CURRENT_BINARY_DIR}/test_find_package_version_${mode}
```
Comment (lrineau) [inline Installation/test/Installation/CMakeLists.txt:129]: `-C$ENV{INIT_FILE}` will create an error if `INIT_FILE` is not in the environment.

Comment (maxGimeno) [reply Installation/test/Installation/CMakeLists.txt:129]: I have seen that. Maybe a genex could help me, I'll dig further...

**Comment by maxGimeno** [comment]
Until local git repos on test machines have been updated, this PR won't pass the  testsuite, so I won't test it again until then.

**Comment by maxGimeno** [comment]
There are some warnings on windows about too long paths.

**Comment by maxGimeno** [comment]
Successfully tested in https://cgal.geometryfactory.com/CGAL/testsuite/results-5.2-Ic-13.shtml

**Comment by lrineau** [comment]
> Successfully tested in [cgal.geometryfactory.com/CGAL/testsuite/results-5.2-Ic-13.shtml](https://cgal.geometryfactory.com/CGAL/testsuite/results-5.2-Ic-13.shtml)

https://cgal.geometryfactory.com/CGAL/testsuite/CGAL-5.2-Ic-13/Installation/TestReport_Blake_Windows_MSVCPreview-Release-64bits.gz
```
Test #43: compile__test_config_file_4 ............***Failed    4.22 sec
```

I will not merge this PR until we find the reason, and fix it.

**Comment by maxGimeno** [comment]
Successfully tested in https://cgal.geometryfactory.com/CGAL/testsuite/results-5.2-Ic-18.shtml

**Pull request merged**